### PR TITLE
Stopped doing hoisting of transitive project dependencies and references

### PR DESCRIPTION
**Pull request by livarcocc**
Stopped doing hoisting of transitive project dependencies and references. We also don't validate if a dependency exist, since migration strictly speaking does not need it. We move the project.json and the csproj will fail to build just like the project.json would.

Fixes https://github.com/dotnet/cli/issues/4959.

@piotrpMSFT @jgoshi @jonsequitur @krwq

### CODE REVIEW

**Comment by livarcocc** [comment]
FYI @333fred

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs [hunk 1]
```diff
@@ -80,34 +80,23 @@ internal class ProjectDependencyFinder
                     projectContext.ProjectFile,
                     framework,
                     preResolvedProjects,
-                    solutionFile,
-                    HoistDependenciesThatAreNotDirectDependencies(projectToResolve, project)
+                    solutionFile
                 );
-                projects.AddRange(dependencies);
```
Comment (jgoshi) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs]: Did you mean to remove this line? Seems like we still need to add dependencies right?

Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs]: Yes, I meant to. This is the line that caused the breath first traversal of the dependency tree to happen.

The line below adds the direct dependencies, which is all we need now.

**Comment by livarcocc** [comment]
@srivatsn

**Pull request merged**